src/library/recording/dlgrecording.cpp

```diff
@@ -129,28 +136,24 @@ void DlgRecording::moveSelection(int delta) {
     m_pTrackTableView->moveSelection(delta);
 }
 
-void DlgRecording::toggleRecording(bool toggle) {
+void DlgRecording::slotRecButtonClicked(bool toggle) {
     Q_UNUSED(toggle);
-    if (!m_pRecordingManager->isRecordingActive()) //If recording is enabled
-    {
-        //pushButtonRecording->setText(tr("Stop Recording"));
-        m_pRecordingManager->startRecording();
-    }
-    else if(m_pRecordingManager->isRecordingActive()) //If we disable recording
-    {
-        //pushButtonRecording->setText(tr("Start Recording"));
-        m_pRecordingManager->stopRecording();
-    }
+    m_pRecordingManager->slotToggleRecording(1);
 }
 
-void DlgRecording::slotRecordingEnabled(bool isRecording) {
+void DlgRecording::slotRecordingStateChanged(bool isRecording) {
     if (isRecording) {
+        pushButtonRecording->setChecked(true);
         pushButtonRecording->setText((tr("Stop Recording")));
```

Comment:
I know you didn't touch the line, but while you're at it:

```suggestion
        pushButtonRecording->setText(tr("Stop Recording"));
```